README.md

```diff
@@ -30,6 +30,37 @@ Note: If dns servers specified on the host or via dns flags are unreachable from
 
 - https://docs.docker.com/config/containers/container-networking/#dns-services
 
+### docker.login
+
+Authenticate the system user `root` against a set of custom docker registries.
```

Comment:
I think this should be noted as optional (as it's only necessary if there's a private registry). 

Something like: "If you need to authenticate against...", also can be moved below the obligatory parts in the README I think.

Reply:
marked this as optional and moved it behind the `docker.machine` substate, which makes `docker.login` the last substate in the list of provided substates